pkgs/desktops/gnome-3/extensions/default.nix

```diff
@@ -0,0 +1,25 @@
+{pkgs, ...}: let
+  # Index of all scraped extensions (with supported versions)
+  extensionsIndex = builtins.fromJSON (builtins.readFile ./extensions.json);
+  # Filter the extensions index to the matching Gnome shell version.
+  filterShellVersion = shell-version: extensions-index:
+    (builtins.map (e: {
+      inherit (e) uuid name description link pname;
+      inherit (e.shell_version_map.${shell-version}) version sha256;
+    }) (builtins.filter
+      (e: (builtins.hasAttr shell-version e."shell_version_map"))
+      extensions-index));
+  # Map a list of derivations to an attribute set with their pname as key
+  derivationsToAttrset = derivations:
+    (builtins.listToAttrs (builtins.map (e: {
+      name = e.pname;
```

Comment:
Use `lib.nameValuePair` instead. It basically is defined as `nameValuePair = name: value: { inherit name; value; }` and makes this 4 line expression a simple 1-line statement.